### PR TITLE
CXSPA-11 [BOPIS] Display Delivery/Pickup Options on Cart page

### PR DESCRIPTION
**Pull request by surya-prakash-singh1**

### CODE REVIEW

File: feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts [hunk 1]
```diff
@@ -48,69 +91,89 @@ export class CartPickupOptionsContainerComponent implements OnInit {
...
-        tap((cart) => (this.cartId = cart.guid ?? '')),
-        tap((cart) => (this.userId = cart.user?.uid ?? ''))
-      )
-      .subscribe();
+    this.displayName$ = outletContext.pipe(
+      map((orderEntry) => ({
+        storeName: orderEntry.deliveryPointOfService?.name,
+        productCode: orderEntry.product.code,
+      })),
+      tap((data) => console.log('data', data)),
```
Comment (cawmichae) [inline feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts]: Remove console.log

File: feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts [hunk 2]
```diff
@@ -48,69 +91,89 @@ export class CartPickupOptionsContainerComponent implements OnInit {
...
+      switchMap(({ storeName, productCode }) =>
+        iif(
+          () => {
+            console.log('storeName', !!storeName);
```
Comment (cawmichae) [inline feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts]: Remove console.log

File: feature-libs/pickup-in-store/core/facade/pickup-option.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  PickupOptionFacade,
+} from '@spartacus/pickup-in-store/root';
+import { Observable } from 'rxjs';
+import {
+  PickupOptionActions,
+  PickupOptionSelectors,
+  StateWithPickupOption,
+} from '../store/index';
+
+// TODO jsdoc
```
Comment (cawmichae) [inline feature-libs/pickup-in-store/core/facade/pickup-option.service.ts]: So do it!

File: feature-libs/pickup-in-store/core/services/preferred-store.service.ts [hunk 4]
```diff
@@ -72,9 +80,35 @@ export class PreferredStoreService implements OnDestroy {
   }
 
   /**
-   * Unsubscribes to dangling subscriptions before destroying the service.
+   * TODO
+   * @param productCode
```
Comment (cawmichae) [inline feature-libs/pickup-in-store/core/services/preferred-store.service.ts]: TODO what?

**Review by cawmichae — APPROVED** [review]

**Review by SVendittelli — APPROVED** [review]

**Pull request merged**